### PR TITLE
Add keywords to filter residues in Janin analysis

### LINKED ISSUE #2898: Janin analysis does not filter CYSH
## Expected behavior ##

The analysis.dihedral.Janin analysis removes known problematic residues. (Or fails with an intelligible error message.)

## Actual behavior ##

When using the test TPR and XTC trajectory, the analysis fails with 

   ValueError: Too many or too few atoms selected. Check for missing or duplicate atoms in topology.

## Code to reproduce the behavior ##

``` python
import MDAnalysis as mda; 
from MDAnalysis.tests import datafiles as data; 
from MDAnalysis.analysis import dihedrals

# GRO works because residue 76 is "CYS"
u = mda.Universe(data.GRO, data.XTC)
J = dihedrals.Janin(u.select_atoms("protein")).run()

# TPR fails because residue 76 is "CYSH"
u2 = mda.Universe(data.TPR, data.XTC)
J2 = dihedrals.Janin(u2.select_atoms("protein")).run()
```

## Current version of MDAnalysis ##

- Which version are you using? (run `python -c "import MDAnalysis as mda; print(mda.__version__)"`) develop
- Which version of Python (`python -V`)? 3.7
- Which operating system? macOS

### PR DESCRIPTION
**Pull request by orbeckst**
Fixes #2898 

Changes made in this Pull Request:
 - filter CYS*
 - updated docs and plots
 - made plots prettier (degree symbol on ticks)

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

File: package/MDAnalysis/analysis/dihedrals.py [hunk 1]
```diff
@@ -99,39 +102,69 @@ class which takes a list of `atomgroups`, this class only needs a list of
...
+~~~~~~~~~~~~~~
+
+Janin plots [Janin1978]_ for side chain conformations (:math:`\chi_1` and
+:math:`chi_2` angles) can be created with the
+:class:`~MDAnalysis.analysis.dihedrals.Janin` class. It works in the same way,
+only needing a list of residues; see the :ref:`Janin plot figure
+<figure-janin>` as an example.
+
+The data for the angles can be accessed in the attribute or
+ :attr:`Janin.angles`.
```
Comment (lilyminium) [inline package/MDAnalysis/analysis/dihedrals.py:127]:
```suggestion
:attr:`Janin.angles`.
```

The space makes this whole line big, like a heading :/

File: package/MDAnalysis/analysis/dihedrals.py [hunk 2]
```diff
@@ -99,39 +102,69 @@ class which takes a list of `atomgroups`, this class only needs a list of
...
+Janin analysis
+~~~~~~~~~~~~~~
+
+Janin plots [Janin1978]_ for side chain conformations (:math:`\chi_1` and
+:math:`chi_2` angles) can be created with the
+:class:`~MDAnalysis.analysis.dihedrals.Janin` class. It works in the same way,
+only needing a list of residues; see the :ref:`Janin plot figure
+<figure-janin>` as an example.
+
+The data for the angles can be accessed in the attribute or
```
Comment (lilyminium) [inline package/MDAnalysis/analysis/dihedrals.py:126]:
```suggestion
The data for the angles can be accessed in the attribute
```

File: package/MDAnalysis/analysis/dihedrals.py [hunk 3]
```diff
@@ -99,39 +102,69 @@ class which takes a list of `atomgroups`, this class only needs a list of
...
+
+Reference plots can be added to the axes for both the Ramachandran and Janin
+classes using the kwarg ``ref=True``. The Ramachandran reference data
+(:data:`~MDAnalysis.analysis.data.filenames.Rama_ref`) and Janin reference data
+(:data:`~MDAnalysis.analysis.data.filenames.Janin_ref`) were made using data
+obtained from a large selection of 500 PDB files, and were analyzed using these
+classes [Mull2018]_. The allowed and marginally allowed regions of the
+Ramachandran reference plot have cutoffs set to include 90% and 99% of the data
+points, and the Janin reference plot has cutoffs for 90% and 98% of the data
+points. The list of PDB files used for the referece plots was taken from
```
Comment (lilyminium) [inline package/MDAnalysis/analysis/dihedrals.py:163]:
```suggestion
points. The list of PDB files used for the reference plots was taken from
```

File: package/MDAnalysis/analysis/dihedrals.py [hunk 4]
```diff
@@ -168,17 +201,30 @@ class which takes a list of `atomgroups`, this class only needs a list of
 
 References
 ----------
+
+.. [Ramachandran1963] G. Ramachandran, C. Ramakrishnan, and
+   V. Sasisekharan. (1963) Stereochemistry of polypeptide chain
+   configurations. *Journal of Molecular Biology*, 7(1):95 – 99. doi:
+   `10.1016/S0022-2836(63)80023-6
+   <https://doi.org/10.1016/S0022-2836(63)80023-6`_
```
Comment (lilyminium) [inline package/MDAnalysis/analysis/dihedrals.py:209]:
```suggestion
   <https://doi.org/10.1016/S0022-2836(63)80023-6>`_
```

File: package/MDAnalysis/analysis/dihedrals.py [hunk 5]
```diff
@@ -372,8 +420,10 @@ def _conclude(self):
         self.angles = np.rad2deg(np.array(self.angles))
 
     def plot(self, ax=None, ref=False, **kwargs):
-        """Plots data into standard ramachandran plot. Each time step in
-        :attr:`Ramachandran.angles` is plotted onto the same graph.
+        """Plots data into standard ramachandran plot.
```
Comment (lilyminium) [inline package/MDAnalysis/analysis/dihedrals.py:423]:
```suggestion
        """Plots data into standard Ramachandran plot.
```

(this was my mistake!)

File: package/MDAnalysis/analysis/dihedrals.py [hunk 6]
```diff
@@ -450,15 +509,15 @@ def __init__(self, atomgroup, **kwargs):
         self.atomgroup = atomgroup
         residues = atomgroup.residues
         protein = atomgroup.universe.select_atoms("protein").residues
```
Comment (lilyminium) [inline package/MDAnalysis/analysis/dihedrals.py:511]:
```suggestion
        protein = atomgroup.select_atoms("protein").residues
```

I would expect the analysis to be limited to whichever atom group I passed in.

**Comment by orbeckst** [comment]
@lilyminium can I please leave it to you to handle the PR? Ping me if I need to do anything else. Thanks!

**Comment by lilyminium** [comment]
Looks good, thank you -- I'll merge now :-)

**Pull request merged**